pubtools/_quay/operator_pusher.py

```diff
@@ -0,0 +1,256 @@
+
+        Args:
+            push_items ([_PushItem]):
+                List of push items.
+            target_name (str):
+                target name
+            target_settings (dict):
+                Target settings.
+        """
+        self.push_items = push_items
+        self.target_settings = target_settings
+
+        self.quay_host = self.target_settings.get("quay_host", "quay.io").rstrip("/")
+
+        self._quay_client = QuayClient(
+            self.target_settings["quay_user"],
+            self.target_settings["quay_password"],
+            self.quay_host,
+        )
+        self._quay_api_client = QuayApiClient(
+            self.target_settings["quay_api_token"], self.quay_host
+        )
+
+    @staticmethod
+    def _get_immutable_tag(push_item):
+        """
+        Return immutable tag from operator push item production tags.
+
+        Args:
+            push_item (_PushItem):
+                Operator push item.
+
+        Returns (str):
+            Immutable tag
+        """
+        # NOTE: while this was copied from pub, '[0]' had to be added here to work properly
+        if push_item.metadata["v_r"] in list(push_item.metadata["tags"].values())[0]:
+            return push_item.metadata["v_r"]
+        # if v_r tag is not in destination tags, return tag with the most numbers in it
+        # This code is usually trigerred when push is not initiated from ET but manually
+        # by user
+        tags = []
+        for tag in list(push_item.metadata["tags"].values())[0]:
+            tags.append((len(re.split(r"\d+", tag)), tag))
+
+        return sorted(tags)[-1][1]
+
+    def public_bundle_ref(self, push_item):
+        """
+        Get public reference of a bundle image.
+
+        It will be used by IIB to access the bundle image.
+
+        Args:
+            push_item (_PushItem):
+                Operator push item.
+
+        Returns (str):
+            Customer-visible bundle reference.
+        """
+        repository = list(push_item.metadata["tags"].keys())[0]
+        # tags are the same for each destination repo, so any combination should work
+        return "{0}/{1}:{2}".format(
+            self.target_settings["docker_reference_registry"][0],
+            repository,
+            self._get_immutable_tag(push_item),
+        )
+
+    def pyxis_get_ocp_versions(self, push_item):
+        """
+        Get a list of supported ocp versions from Pyxis.
+
+        Args:
+            push_item: (_PushItem)
+                Push item for which the OCP version range will be found out.
+
+        Returns ([str]):
+            Supported OCP versions as returned by Pyxis.
+        """
+        ocp_versions = push_item.metadata["com.redhat.openshift.versions"]
+        LOG.info("Getting OCP versions of '{0}' from Pyxis.".format(ocp_versions))
+
+        args = ["--pyxis-server", self.target_settings["pyxis_server"]]
+        args = ["--pyxis-krb-principal", self.target_settings["iib_krb_principal"]]
```

Comment:
Looks like a typo — should be `+=`? If so, the tests probably need to be tweaked to make sure this is covered.

Reply:
Great catch. The tests do catch this, but the expected value was incorrect as well :facepalm: